### PR TITLE
Filter active teachers and coordinators in status_setter

### PR DESCRIPTION
**Pull request by Roxe322**
closes (part of?) #158

### CODE REVIEW

File: api/models/choices/status.py [hunk 1]
```diff
@@ -33,6 +33,18 @@ class CoordinatorStatus(models.TextChoices):
...
+    @classmethod
+    def active_statuses(cls) -> list["CoordinatorStatus"]:
+        return [
+            cls.PENDING,
+            cls.ONBOARDING,
+            cls.ONBOARDING_STALE,
+            cls.WORKING_BELOW_THRESHOLD,
+            cls.WORKING_OK,
+            cls.WORKING_LIMIT_REACHED,
+            cls.FINISHED_LEFT,
```
Comment (ivanBarinsky) [inline api/models/choices/status.py:45]: Not sure if thats correct?

Comment (ivanBarinsky) [reply api/models/choices/status.py:45]: I mean just the one on line 45, sounds like if a coordinator has that status, they arent really part of the project anymore

Comment (lemontree210) [inline api/models/choices/status.py:45]:
```suggestion
```

Comment (lemontree210) [reply api/models/choices/status.py:45]: Yep, I'm adding these notes 👌🏻

File: api/models/choices/status.py [hunk 2]
```diff
@@ -33,6 +33,18 @@ class CoordinatorStatus(models.TextChoices):
     REMOVED = "removed", "All access revoked, accounts closed"
     BANNED = "banned", "Banned from the project"
 
+    @classmethod
+    def active_statuses(cls) -> list["CoordinatorStatus"]:
+        return [
+            cls.PENDING,
+            cls.ONBOARDING,
+            cls.ONBOARDING_STALE,
```
Comment (lemontree210) [inline api/models/choices/status.py:41]:
```suggestion
```

File: api/models/choices/status.py [hunk 3]
```diff
@@ -97,6 +109,21 @@ class TeacherStatus(models.TextChoices):
     BANNED = "banned", "Banned from the project"
     REMOVED = "removed", "All access revoked, accounts closed"
 
+    @classmethod
+    def active_statuses(cls) -> list["TeacherStatus"]:
+        return [
+            cls.AWAITING_OFFER,
+            cls.GROUP_OFFERED,
+            cls.AWAITING_START,
```
Comment (lemontree210) [inline api/models/choices/status.py:117]:
```suggestion
```

File: api/models/choices/status.py [hunk 4]
```diff
@@ -97,6 +109,21 @@ class TeacherStatus(models.TextChoices):
...
+            cls.TEACHING_ACCEPTING_MORE,
+            cls.TEACHING_NOT_ACCEPTING_MORE,
+            cls.TEACHING_ANOTHER_GROUP_OFFERED,
+            cls.TEACHING_AWAITING_START_OF_ANOTHER,
+            cls.NEEDS_SUBSTITUTION,
+            # TODO: need to confirm this is correct
+            cls.FINISHED_STAYS,
```
Comment (lemontree210) [inline api/models/choices/status.py:124]:
```suggestion
```

File: api/models/choices/status.py [hunk 5]
```diff
@@ -113,5 +140,12 @@ class TeacherUnder18Status(models.TextChoices):
     BANNED = "banned", "Banned from the project"
     REMOVED = "removed", "All access revoked, accounts closed"
 
+    @classmethod
+    def active_statuses(cls) -> list["TeacherUnder18Status"]:
+        return [
+            cls.ACTIVE,
+            cls.FINISHED_STAYS,
```
Comment (lemontree210) [inline api/models/choices/status.py:147]:
```suggestion
```

**Review by lemontree210 — COMMENTED** [review]
I think that this PR uncovered a flaw in how I put the business logic into the statuses. I think I mixed two types of statuses:
1. permanent status in the project (pending, working, banned, finished_left, finished_stays)
2. temporary status caused by some event (group_offered, awaiting_start, requested_transfer).

Hence those ugly statuses like "working_awaiting_start_of_another". Hence it's hard for me right now to judge which statuses should be deemed "active".

I think that I will need a day or two to figure out the best way of dealing with this. Maybe splitting the statuses will be a good thing because it would better reflect the business logic.

**Comment by Auch-Auch** [comment]
Just a small suggestion from me, usually it is a good idea to simplify statuses at the start and keep only mandatory ones. After we will have real users it will be more obvious what is actually needed.

**Review by lemontree210 — APPROVED** [review]

**Comment by lemontree210** [comment]
Thank you!

**Pull request merged**